C/common/management_client.cpp

```diff
@@ -2059,3 +2059,67 @@ int ManagementClient::validateDatapoints(std::string dp1, std::string dp2)
 
 	return temp.compare(dp2);
 }
+
+/**
+ * Get an alert by specific key
+ *
+ * @param    key        Key to get alert
+ * @return   string     Alert
+ */
+std::string ManagementClient::getAlertByKey(const std::string& key)
+{
+	std::string response = "Status: 404 Not found";
+	try
+	{
+		std::string url = "/fledge/alert/" + urlEncode(key) ;
+		auto res = this->getHttpClient()->request("GET", url.c_str());
+		Document doc;
+		response = res->content.string();
+		doc.Parse(response.c_str());
+		if (doc.HasParseError())
+		{
+			bool httpError = (isdigit(response[0]) && isdigit(response[1]) && isdigit(response[2]) && response[3]==':');
```

Comment:
You don't need to do this. It is best to check res->status_code before trying to parse the GET response.